pkg/tsdb/cloudwatch/services/regions_test.go

```diff
@@ -0,0 +1,62 @@
+package services
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models/resources"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type EC2Mock struct {
+	mock.Mock
+}
+
+func (e *EC2Mock) DescribeRegions(in *ec2.DescribeRegionsInput) (*ec2.DescribeRegionsOutput, error) {
+	args := e.Called()
+	mockResponse := args.Get(0)
+	if mockResponse == nil {
+		return nil, args.Error(1)
+	}
+	return mockResponse.(*ec2.DescribeRegionsOutput), args.Error(1)
+}
+
```

Comment:
This type can be in `mocks` if you want to match the refactoring.
Any reason for the logic in the method there?

Reply:
cool moving it to mocks, sadly I get the same nil error as above if I remove the conditional :/